opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -292,6 +295,20 @@ impl LogProcessor for BatchLogProcessor {
                 otel_warn!(name: "BatchLogProcessor.LogDroppingStarted",
                     message = "BatchLogProcessor dropped a LogRecord due to queue full/internal errors. No further log will be emitted for further drops until Shutdown. During Shutdown time, a log will be emitted with exact count of total logs dropped.");
             }
+            return;
+        }
+
+        // At this point, sending the log record to the data channel was successful.
+        // Increment the current batch size and check if it has reached the max export batch size.
+        if self.current_batch_size.fetch_add(1, Ordering::Relaxed) >= self.max_export_batch_size {
```

Comment:
should this be changed to
```suggestion
        if self.current_batch_size.fetch_add(1, Ordering::Relaxed) + 1 >= self.max_export_batch_size {
```


as `fetch_add` returns value before increment. So as per the pr logic, if `max_export_batch_size` = 2000, the export signal would be sent when `current_batch_size` = 2001, which doesn't seem correct.
`

Reply:
Thanks for catching that!